tools/d2g/d2gtest/d2g_test.go

```diff
@@ -197,6 +175,88 @@ func validateTestSuite(t *testing.T, schema string, path string) {
 
 func (tc *TestCase) Validate(t *testing.T) {
 	t.Helper()
-	validateAgainstSchema(t, tc.DockerWorkerTaskPayload, dockerworker.JSONSchema())
-	validateAgainstSchema(t, tc.GenericWorkerTaskPayload, genericworker.JSONSchema())
+	var dwRaw json.RawMessage
+	var gwRaw json.RawMessage
+
+	if tc.IsTaskDef {
+		var parsedDwTaskDef map[string]interface{}
+		err := json.Unmarshal(tc.DockerWorkerTaskDefinition, &parsedDwTaskDef)
+		if err != nil {
+			t.Fatalf("cannot parse Docker Worker task definition: %v", err)
+		}
+
+		dwRaw, err = json.Marshal(parsedDwTaskDef["payload"])
+		if err != nil {
+			t.Fatalf("Cannot marshal test suite Docker Worker payload %#v: %v", parsedDwTaskDef["payload"], err)
+		}
+
+		var parsedGwTaskDef map[string]interface{}
+		err = json.Unmarshal(tc.GenericWorkerTaskDefinition, &parsedGwTaskDef)
+		if err != nil {
+			t.Fatalf("cannot parse Generic Worker task definition: %v", err)
+		}
+
+		gwRaw, err = json.Marshal(parsedGwTaskDef["payload"])
+		if err != nil {
+			t.Fatalf("Cannot marshal test suite Generic Worker payload %#v: %v", parsedGwTaskDef["payload"], err)
+		}
+	} else {
+		dwRaw = tc.DockerWorkerTaskPayload
+		gwRaw = tc.GenericWorkerTaskPayload
+	}
```

Comment:
It feels like we should have two different types for a PayloadTestCase and a TaskDefinitionTestCase, which I think sits well with having a `oneOf` too.